tests/components/unifi/test_sensor.py

```diff
@@ -183,6 +185,37 @@ async def test_bandwidth_sensors(
     assert hass.states.get("sensor.wired_client_rx") is None
     assert hass.states.get("sensor.wired_client_tx") is None
 
+    # Enable option
+
+    options[CONF_ALLOW_BANDWIDTH_SENSORS] = True
+    hass.config_entries.async_update_entry(config_entry, options=options.copy())
+    await hass.async_block_till_done()
+
+    assert len(hass.states.async_all()) == 5
+    assert len(hass.states.async_entity_ids(SENSOR_DOMAIN)) == 4
+    assert hass.states.get("sensor.wireless_client_rx")
+    assert hass.states.get("sensor.wireless_client_tx")
+    assert hass.states.get("sensor.wired_client_rx")
+    assert hass.states.get("sensor.wired_client_tx")
+
+    # Try to add the sensors again, using a signal
+
+    clients_connected = {wired_client["mac"], wireless_client["mac"]}
+    devices_connected = set()
+
+    controller = hass.data[UNIFI_DOMAIN][config_entry.entry_id]
+
+    async_dispatcher_send(
```

Comment:
Side note: A more robust approach would be to get access to the callback set on the client api and use that callback to trigger the dispatch indirectly.

Reply:
Thanks Martin! This is a legacy signal which I aim to remove once the currently open PR for refactoring UniFi integration is completed.

Reply:
Can't we use the `mock_unifi_websocket` fixture to trigger the callback?

Reply:
Yes. But as it's planned to be removed I'd rather focus on getting to that point

Reply:
Is the websocket planned to be removed? The point is that we want to write tests that don't need to be changed when we refactor an integration.

Reply:
I get your point

Reply:
Changed here https://github.com/home-assistant/core/pull/89525 control is owned by library so removed that part of the test.